index.js

```diff
@@ -1,5 +1,3 @@
-var toString = {}.toString;
-
 module.exports = Array.isArray || function (arr) {
-  return toString.call(arr) === '[object Array]';
+  return Object.toString.call(arr) === '[object Array]';
```

Comment:
This is a very bad change; it means the function won’t be cached in the module - which protects against later mutation - and also, the function is on Object.prototype, not Object - this change uses Function toString, which would break every consumer of this module.

Reply:
@ljharb First of all, sorry for the omission of prototype.

But why cache the toString? 

I don't think it needs to be cached, and caching adds unnecessary memory consumption.

Reply:
it ABSOLUTELY needs to be cached, so that later-running code that does `delete Object.prototype.toString` doesn't break this package.

Memory is infinite and free compared to the cost of bugs.